# Known divergence (harness artifact): the regression suite expects to run in a
# database named "regression", but multigateway cannot CREATE DATABASE, so the
# harness runs it in "postgres". Every difference below is the catalog name
# column of information_schema foreign-data views reporting "postgres" instead
# of "regression"; the foreign-data wrapper, server, user-mapping and foreign
# table behavior itself matches upstream.
--- a
+++ b
@@ -969,122 +969,122 @@
 SELECT * FROM information_schema.foreign_data_wrappers ORDER BY 1, 2;
 foreign_data_wrapper_catalog | foreign_data_wrapper_name | authorization_identifier | library_name | foreign_data_wrapper_language
 ------------------------------+---------------------------+---------------------------+--------------+-------------------------------
-regression | dummy | regress_foreign_data_user | | c
-regression | foo | regress_foreign_data_user | | c
-regression | postgresql | regress_foreign_data_user | | c
+postgres | dummy | regress_foreign_data_user | | c
+postgres | foo | regress_foreign_data_user | | c
+postgres | postgresql | regress_foreign_data_user | | c
 (3 rows)
 
 SELECT * FROM information_schema.foreign_data_wrapper_options ORDER BY 1, 2, 3;
 foreign_data_wrapper_catalog | foreign_data_wrapper_name | option_name | option_value
 ------------------------------+---------------------------+--------------+--------------
-regression | foo | test wrapper | true
+postgres | foo | test wrapper | true
 (1 row)
 
 SELECT * FROM information_schema.foreign_servers ORDER BY 1, 2;
 foreign_server_catalog | foreign_server_name | foreign_data_wrapper_catalog | foreign_data_wrapper_name | foreign_server_type | foreign_server_version | authorization_identifier
 ------------------------+---------------------+------------------------------+---------------------------+---------------------+------------------------+---------------------------
-regression | s0 | regression | dummy | | | regress_foreign_data_user
-regression | s4 | regression | foo | oracle | | regress_foreign_data_user
-regression | s5 | regression | foo | | 15.0 | regress_test_role
-regression | s6 | regression | foo | | 16.0 | regress_test_indirect
-regression | s8 | regression | postgresql | | | regress_foreign_data_user
-regression | t1 | regression | foo | | | regress_test_indirect
-regression | t2 | regression | foo | | | regress_test_role
+postgres | s0 | postgres | dummy | | | regress_foreign_data_user
+postgres | s4 | postgres | foo | oracle | | regress_foreign_data_user
+postgres | s5 | postgres | foo | | 15.0 | regress_test_role
+postgres | s6 | postgres | foo | | 16.0 | regress_test_indirect
+postgres | s8 | postgres | postgresql | | | regress_foreign_data_user
+postgres | t1 | postgres | foo | | | regress_test_indirect
+postgres | t2 | postgres | foo | | | regress_test_role
 (7 rows)
 
 SELECT * FROM information_schema.foreign_server_options ORDER BY 1, 2, 3;
 foreign_server_catalog | foreign_server_name | option_name | option_value
 ------------------------+---------------------+-----------------+--------------
-regression | s4 | dbname | b
-regression | s4 | host | a
-regression | s6 | dbname | b
-regression | s6 | host | a
-regression | s8 | connect_timeout | 30
-regression | s8 | dbname | db1
+postgres | s4 | dbname | b
+postgres | s4 | host | a
+postgres | s6 | dbname | b
+postgres | s6 | host | a
+postgres | s8 | connect_timeout | 30
+postgres | s8 | dbname | db1
 (6 rows)
 
 SELECT * FROM information_schema.user_mappings ORDER BY lower(authorization_identifier), 2, 3;
 authorization_identifier | foreign_server_catalog | foreign_server_name
 ---------------------------+------------------------+---------------------
-PUBLIC | regression | s4
-PUBLIC | regression | s8
-PUBLIC | regression | t1
-regress_foreign_data_user | regression | s4
-regress_foreign_data_user | regression | s8
-regress_test_role | regression | s5
-regress_test_role | regression | s6
-regress_test_role | regression | t1
+PUBLIC | postgres | s4
+PUBLIC | postgres | s8
+PUBLIC | postgres | t1
+regress_foreign_data_user | postgres | s4
+regress_foreign_data_user | postgres | s8
+regress_test_role | postgres | s5
+regress_test_role | postgres | s6
+regress_test_role | postgres | t1
 (8 rows)
 
 SELECT * FROM information_schema.user_mapping_options ORDER BY lower(authorization_identifier), 2, 3, 4;
 authorization_identifier | foreign_server_catalog | foreign_server_name | option_name | option_value
 ---------------------------+------------------------+---------------------+--------------+--------------
-PUBLIC | regression | s4 | this mapping | is public
-PUBLIC | regression | t1 | modified | 1
-regress_foreign_data_user | regression | s8 | password | public
-regress_test_role | regression | s5 | modified | 1
-regress_test_role | regression | s6 | username | test
-regress_test_role | regression | t1 | password | boo
-regress_test_role | regression | t1 | username | bob
+PUBLIC | postgres | s4 | this mapping | is public
+PUBLIC | postgres | t1 | modified | 1
+regress_foreign_data_user | postgres | s8 | password | public
+regress_test_role | postgres | s5 | modified | 1
+regress_test_role | postgres | s6 | username | test
+regress_test_role | postgres | t1 | password | boo
+regress_test_role | postgres | t1 | username | bob
 (7 rows)
 
 SELECT * FROM information_schema.usage_privileges WHERE object_type LIKE 'FOREIGN%' AND object_name IN ('s6', 'foo') ORDER BY 1, 2, 3, 4, 5;
 grantor | grantee | object_catalog | object_schema | object_name | object_type | privilege_type | is_grantable
 ---------------------------+---------------------------+----------------+---------------+-------------+----------------------+----------------+--------------
-regress_foreign_data_user | regress_foreign_data_user | regression | | foo | FOREIGN DATA WRAPPER | USAGE | YES
-regress_foreign_data_user | regress_test_indirect | regression | | foo | FOREIGN DATA WRAPPER | USAGE | NO
-regress_test_indirect | regress_test_indirect | regression | | s6 | FOREIGN SERVER | USAGE | YES
-regress_test_indirect | regress_test_role2 | regression | | s6 | FOREIGN SERVER | USAGE | YES
+regress_foreign_data_user | regress_foreign_data_user | postgres | | foo | FOREIGN DATA WRAPPER | USAGE | YES
+regress_foreign_data_user | regress_test_indirect | postgres | | foo | FOREIGN DATA WRAPPER | USAGE | NO
+regress_test_indirect | regress_test_indirect | postgres | | s6 | FOREIGN SERVER | USAGE | YES
+regress_test_indirect | regress_test_role2 | postgres | | s6 | FOREIGN SERVER | USAGE | YES
 (4 rows)
 
 SELECT * FROM information_schema.role_usage_grants WHERE object_type LIKE 'FOREIGN%' AND object_name IN ('s6', 'foo') ORDER BY 1, 2, 3, 4, 5;
 grantor | grantee | object_catalog | object_schema | object_name | object_type | privilege_type | is_grantable
 ---------------------------+---------------------------+----------------+---------------+-------------+----------------------+----------------+--------------
-regress_foreign_data_user | regress_foreign_data_user | regression | | foo | FOREIGN DATA WRAPPER | USAGE | YES
-regress_foreign_data_user | regress_test_indirect | regression | | foo | FOREIGN DATA WRAPPER | USAGE | NO
-regress_test_indirect | regress_test_indirect | regression | | s6 | FOREIGN SERVER | USAGE | YES
-regress_test_indirect | regress_test_role2 | regression | | s6 | FOREIGN SERVER | USAGE | YES
+regress_foreign_data_user | regress_foreign_data_user | postgres | | foo | FOREIGN DATA WRAPPER | USAGE | YES
+regress_foreign_data_user | regress_test_indirect | postgres | | foo | FOREIGN DATA WRAPPER | USAGE | NO
+regress_test_indirect | regress_test_indirect | postgres | | s6 | FOREIGN SERVER | USAGE | YES
+regress_test_indirect | regress_test_role2 | postgres | | s6 | FOREIGN SERVER | USAGE | YES
 (4 rows)
 
 SELECT * FROM information_schema.foreign_tables ORDER BY 1, 2, 3;
 foreign_table_catalog | foreign_table_schema | foreign_table_name | foreign_server_catalog | foreign_server_name
 -----------------------+----------------------+--------------------+------------------------+---------------------
-regression | foreign_schema | foreign_table_1 | regression | s0
+postgres | foreign_schema | foreign_table_1 | postgres | s0
 (1 row)
 
 SELECT * FROM information_schema.foreign_table_options ORDER BY 1, 2, 3, 4;
 foreign_table_catalog | foreign_table_schema | foreign_table_name | option_name | option_value
 -----------------------+----------------------+--------------------+-------------+--------------
-regression | foreign_schema | foreign_table_1 | be quoted | value
-regression | foreign_schema | foreign_table_1 | escape | @
-regression | foreign_schema | foreign_table_1 | quote | ~
+postgres | foreign_schema | foreign_table_1 | be quoted | value
+postgres | foreign_schema | foreign_table_1 | escape | @
+postgres | foreign_schema | foreign_table_1 | quote | ~
 (3 rows)
 
 SET ROLE regress_test_role;
 SELECT * FROM information_schema.user_mapping_options ORDER BY 1, 2, 3, 4;
 authorization_identifier | foreign_server_catalog | foreign_server_name | option_name | option_value
 --------------------------+------------------------+---------------------+-------------+--------------
-PUBLIC | regression | t1 | modified | 1
-regress_test_role | regression | s5 | modified | 1
-regress_test_role | regression | s6 | username | test
-regress_test_role | regression | t1 | password | boo
-regress_test_role | regression | t1 | username | bob
+PUBLIC | postgres | t1 | modified | 1
+regress_test_role | postgres | s5 | modified | 1
+regress_test_role | postgres | s6 | username | test
+regress_test_role | postgres | t1 | password | boo
+regress_test_role | postgres | t1 | username | bob
 (5 rows)
 
 SELECT * FROM information_schema.usage_privileges WHERE object_type LIKE 'FOREIGN%' AND object_name IN ('s6', 'foo') ORDER BY 1, 2, 3, 4, 5;
 grantor | grantee | object_catalog | object_schema | object_name | object_type | privilege_type | is_grantable
 ---------------------------+-----------------------+----------------+---------------+-------------+----------------------+----------------+--------------
-regress_foreign_data_user | regress_test_indirect | regression | | foo | FOREIGN DATA WRAPPER | USAGE | NO
-regress_test_indirect | regress_test_indirect | regression | | s6 | FOREIGN SERVER | USAGE | YES
-regress_test_indirect | regress_test_role2 | regression | | s6 | FOREIGN SERVER | USAGE | YES
+regress_foreign_data_user | regress_test_indirect | postgres | | foo | FOREIGN DATA WRAPPER | USAGE | NO
+regress_test_indirect | regress_test_indirect | postgres | | s6 | FOREIGN SERVER | USAGE | YES
+regress_test_indirect | regress_test_role2 | postgres | | s6 | FOREIGN SERVER | USAGE | YES
 (3 rows)
 
 SELECT * FROM information_schema.role_usage_grants WHERE object_type LIKE 'FOREIGN%' AND object_name IN ('s6', 'foo') ORDER BY 1, 2, 3, 4, 5;
 grantor | grantee | object_catalog | object_schema | object_name | object_type | privilege_type | is_grantable
 ---------------------------+-----------------------+----------------+---------------+-------------+----------------------+----------------+--------------
-regress_foreign_data_user | regress_test_indirect | regression | | foo | FOREIGN DATA WRAPPER | USAGE | NO
-regress_test_indirect | regress_test_indirect | regression | | s6 | FOREIGN SERVER | USAGE | YES
-regress_test_indirect | regress_test_role2 | regression | | s6 | FOREIGN SERVER | USAGE | YES
+regress_foreign_data_user | regress_test_indirect | postgres | | foo | FOREIGN DATA WRAPPER | USAGE | NO
+regress_test_indirect | regress_test_indirect | postgres | | s6 | FOREIGN SERVER | USAGE | YES
+regress_test_indirect | regress_test_role2 | postgres | | s6 | FOREIGN SERVER | USAGE | YES
 (3 rows)
 
 DROP USER MAPPING FOR current_user SERVER t1;
@@ -1092,7 +1092,7 @@
 SELECT * FROM information_schema.user_mapping_options ORDER BY 1, 2, 3, 4;
 authorization_identifier | foreign_server_catalog | foreign_server_name | option_name | option_value
 --------------------------+------------------------+---------------------+-------------+--------------
-regress_test_role | regression | s6 | username |
+regress_test_role | postgres | s6 | username |
 (1 row)
 
 RESET ROLE;
